test/audit-log/mzcompose.py

```diff
@@ -0,0 +1,43 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import time
+
+from materialize.mzcompose import Composition
+from materialize.mzcompose.services import Materialized, Testdrive
+
+SERVICES = [
+    Materialized(),
+    Testdrive(),
+]
+
+
+def workflow_default(c: Composition) -> None:
+    c.up("materialized")
+    c.wait_for_materialized(service="materialized")
+
+    # Create some audit log entries.
+    c.sql("CREATE TABLE t (i INT)")
+    c.sql("CREATE DEFAULT INDEX ON t")
+
+    # occurred_at is a timestamp with microsecond precesion, so order by something
+    # else also to avoid flakes in case two events share a timestamp.
+    log = c.query("SELECT * FROM mz_audit_log ORDER BY occurred_at, uuid")
+
+    # Restart mz.
+    c.kill("materialized")
+    c.up("materialized")
+    c.wait_for_materialized()
+
+    # Verify the audit log entries are still present and have not changed.
+    restart_log = c.query("SELECT * FROM mz_audit_log ORDER BY occurred_at, uuid")
+    if log != restart_log:
+        print("initial audit log:", log)
+        print("audit log after restart:", restart_log)
+        raise Exception("audit logs not equal after restart")
```

Comment:
This is a good candidate for test/restart rather than its own mzcompose composition. (Each composition has a huge fixed cost in CI to download the Docker images onto their own agent.)

Reply:
Done.